state.js

```diff
@@ -48,27 +49,32 @@ function readState () {
   var str = location.pathname.substr(1)
   var result = getRouteNode(str)
 
-  if (!result)
-    return undefined
+  if (!result) {
+    const ev = new CustomEvent("404", {
+      detail: {
+        path: str
+      }
+    })
+
+    window.dispatchEvent(ev)
+    return
+  }
   var node = cloneNodeAsElement(result.node, 'div')
 
   node.innerHTML = result.node.textContent
   loadNodeSource(node, result.matches)
   return node
 }
 
-/**
- * Starts listening to the "popstate" event after page loads and fires initial
- * change state for the page.
- */
-function startState () {
-  function on (e) {
-    changeState(location.pathname + location.search, e.state, '')
+function onPopState (e) {
+  changeState(location.pathname + location.search, e.state, '')
+}
+
+function startState (e) {
```

Comment:
I've opted to remove this method all together. Developers should implement their own way to add onPopState in their own DOMContentLoaded listener. `startState` has caused more issues for us than need be.